core/app/components/Memberships/RoleSelect.tsx

```diff
@@ -0,0 +1,83 @@
+"use client";
+
+import type { UsersId } from "db/public";
+import { MemberRole } from "db/public";
+import { Button } from "ui/button";
+import {
+	DropdownMenu,
+	DropdownMenuContent,
+	DropdownMenuItem,
+	DropdownMenuTrigger,
+} from "ui/dropdown-menu";
+import { toast } from "ui/use-toast";
+
+import type { MembersListProps, TargetId } from "./types";
+import { didSucceed, useServerAction } from "~/lib/serverActions";
+
+type RoleSelectProps<T extends TargetId> = {
+	userId: UsersId;
+	targetId: T;
+	setRole: MembersListProps<T>["setRole"];
+	role: MemberRole;
+};
+
+export const RoleSelect = <T extends TargetId>({
+	targetId,
+	role: currentRole,
+	userId,
+	setRole,
+}: RoleSelectProps<T>) => {
+	return (
+		<DropdownMenu>
+			<DropdownMenuTrigger asChild>
+				<Button className="rounded-full bg-blue-500 capitalize text-slate-50 hover:bg-blue-600">
```

Comment:
again tiny tiny thing, but i think it'd look better if it was a small button instead

```suggestion
				<Button
					size="sm"
					className="rounded-full bg-blue-500 capitalize text-slate-50 hover:bg-blue-600"
				>
```